aten/src/ATen/native/Sorting.cpp

```diff
@@ -100,7 +100,7 @@ void _fill_indices(const TensorBase &indices, int64_t dim) {
   auto ndim = indices.dim();
   assert(0 <= dim && dim < ndim);
   auto dim_size = indices.size(dim);
-  auto idx_dim = at::arange(0, dim_size, indices.options().dtype(at::kLong));
+  auto idx_dim = at::arange(0, dim_size, indices.options());
```

Comment:
This still needs a dtype long if the dimsize is large enough, doesn't it?

Reply:
Otherwise, it will not float64 for arange incremental kernel and may cause inaccuracy issues above 14 million elements